src/main/java/slp/core/example/JavaRunner.java

```diff
@@ -26,50 +26,62 @@ public static void main(String[] args) {
-		//		should yield the same result, but could e.g. allow one to write the vocabulary before training.
-		//    - If interested, use: VocabularyRunner.build(train);
+		//		should yield the same result, and may be useful if you want to write the vocabulary before training.
+		//    - If interested, use: VocabularyRunner.build(lexer, train);
+		
 		
 		// 3. Model
-		//    a. No per line modeling for Java (default)
-		ModelRunner.perLine(false);
-		//    b. Self-testing if train is equal to test; will un-count each file before modeling it.
-		ModelRunner.selfTesting(train.equals(test));
-		//    c. Set n-gram model order, 6 works well for Java
-		ModelRunner.setNGramOrder(6);
-		//    d. Use an n-gram model with simple Jelinek-Mercer smoothing (works well for code)
-		//       - Let's use a GigaCounter, fit for large corpora here as well; the nested model later on will copy this behavior.
-		Model model = new JMModel(new GigaCounter());
-		//    e. First, train this model on all files in 'train' recursively, using the usual updating mechanism (same as for dynamic updating).
+		//	  a. We will use an n-gram model with simple Jelinek-Mercer smoothing (works well for code)
+		//		 - The n-gram order of 6 is used, which is also the standard
+		//       - Let's use a GigaCounter (useful for large corpora) here as well; the nested model later on will copy this behavior.
+		Model model = new JMModel(6, new GigaCounter());
+		//       - We pass this Model to a ModelRunner, which takes care of most of the work and configuration for us
+		ModelRunner modelRunner = new ModelRunner(lexerRunner, model);
+		//    b. Self-testing if train is equal to test; this will un-count each file before modeling it.
+		modelRunner.setSelfTesting(train.equals(test));
+		//    c. First, train this model on all files in 'train' recursively, using the usual updating mechanism (same as for dynamic updating).
 		//       - Note that this invokes Model.learn for each file, which is fine for n-gram models since these are count-based;
 		//          other models may prefer to pre-train when calling the Model's constructor.
-		ModelRunner.learn(model, train);
-		//    f. To get more fancy for source code, we can convert the model into a complex mix model.
+		modelRunner.learn(train);
+		//    d. To get more fancy for source code, we can convert the model into a complex mix model.
 		//       - First wrap in a nested model, causing it to learn all files in test into a new 'local' model
-		model = new NestedModel(test, model);
+		modelRunner.setNested(test);
 		//       - Then, add an ngram-cache component.
 		//         * Note, order matters! The most local model should be "right-most" so it is called upon last (i.e. "gets the final say")
 		//         * This is also why we apply the cache after the nested model
-		model = new InverseMixModel(model, new CacheModel());
+		model = MixModel.standard(model, new CacheModel());
```

Comment:
I'm confused about how the `CacheModel` mixture will be used. It doesn't seem to be referenced by the `ModelRunner` or accessed directly?

Reply:
The cache model learns on the fly, meaning that it integrates every event only after being asked to model it. What facilitates this is that it is `dynamic` in nature; other models can get the same benefit by setting themselves to dynamic as well. The `AbstractModel` takes care of invoking the learning after modeling.

Reply:
As far as I can tell, though, the mix-model is being constructed after the model-runner is, and so the object containing the cache is never being invoked in this example.

Reply:
You're right, I see it now. I need to think about how to resolve this, since you shouldn't have to first compose the whole model before passing it to a `ModelRunner`. Maybe it should be a settable field, or not a field at all. I'll hold off on this until I see a more fitting solution.